lib/phoenix/router.ex

```diff
@@ -1282,4 +1298,62 @@ defmodule Phoenix.Router do
       Map.delete(metadata, :conn)
     end
   end
+
+  def __formatted_routes__(router) do
+    Enum.flat_map(router.__routes__(), fn route ->
+      Code.ensure_loaded(route.plug)
+
+      if function_exported?(route.plug, :formatted_routes, 1) do
+        route.plug_opts
+        |> route.plug.formatted_routes()
+        |> Enum.map(fn nested_route ->
+          %{
+            route
+            | path: Path.join(route.path, nested_route.path),
+              verb: nested_route.verb,
+              plug_opts: nested_route.plug_opts
+          }
+        end)
+      else
+        plug =
+          case route.metadata[:mfa] do
+            {module, _, _} -> module
+            _ -> route.plug
+          end
+
+        label = "#{inspect(plug)} #{inspect(route.plug_opts)}"
+
+        [
+          %{
+            helper: route.helper,
+            verb: route.verb,
+            path: route.path,
+            label: label
+          }
+        ]
+      end
+    end)
+  end
+
+  def __verified_route__?(router, split_path) do
+    case router.__verify_route__(split_path) do
+      {_forward_plug, true = _warn_on_verify?} ->
+        false
+
+      {nil = _forward_plug, false = _warn_on_verify?} ->
+        true
+
+      {{router, script_name, plug_opts}, false = _warn_on_verify?} ->
+        Code.ensure_loaded(router)
+
+        if function_exported?(router, :verified_route?, 2) do
+          router.verified_route?(plug_opts, split_path -- script_name)
+        else
+          false
```

Comment:
As per the current implementation:

```suggestion
          true
```